### PR TITLE
updated popper.js, added 'hide' modifier to Tooltip & Dropdown

### PR DESCRIPTION
**Pull request by dsv-sergey**
- [Breaking Change]: In Popper 2 this is now an array of objects (modifiers), instead of an object where each property was the modifier name. [More information.](https://popper.js.org/docs/v2/migration-guide/)

### CODE REVIEW

File: uui-components/src/overlays/Dropdown.tsx [hunk 1]
```diff
@@ -41,7 +40,7 @@ export interface DropdownProps extends Partial<IEditable<boolean>> {
     closeOnMouseLeave?: 'toggler' | 'boundary' | false;
 
     portalTarget?: HTMLElement;
-    boundaryElement?: PopperJS.Boundary | Element;
+    boundaryElement?: Boundary;
```
Comment (AlekseyManetov) [inline uui-components/src/overlays/Dropdown.tsx:43]: In Popper 2 'Element' can't be passed?

File: uui-components/src/overlays/Dropdown.tsx [hunk 2]
```diff
@@ -183,17 +180,39 @@ export class Dropdown extends React.Component<DropdownProps, DropdownState> {
         );
     }
 
-    private isInteractedOutside = (e: Event) => {
-        if (!this.isOpened()) {
-            return false;
+    private renderChildren = ({ ref, placement, style, update, isReferenceHidden }: PopperChildrenProps) => {
```
Comment (AlekseyManetov) [inline uui-components/src/overlays/Dropdown.tsx:183]: Maybe gives more specific name, like 'renderDropdownBody'?

File: uui-components/src/overlays/Dropdown.tsx [hunk 3]
```diff
@@ -183,17 +180,39 @@ export class Dropdown extends React.Component<DropdownProps, DropdownState> {
...
+        if (isReferenceHidden && this.state.opened) {
+            setTimeout(() => { this.handleOpenedChange(false); }, 0);
+            return null;
         }
+        return (
+            <div
+                className='uui-popper'
+                ref={ node => {
```
Comment (AlekseyManetov) [inline uui-components/src/overlays/Dropdown.tsx:191]: Can you move it in the separate function too

File: uui-components/src/overlays/Tooltip.tsx [hunk 4]
```diff
@@ -1,23 +1,24 @@
...
     renderContent?(): any;
-    placement?: PopperJS.Placement;
+    placement?: Placement;
     trigger?: 'click' | 'press' | 'hover';
     portalTarget?: HTMLElement;
-    offset?: string;
+    offset?: Options['offset'];
     children?: React.ReactNode;
-    boundaryElement?: Boundary | Element;
+    boundaryElement?: Boundary;
```
Comment (AlekseyManetov) [inline uui-components/src/overlays/Tooltip.tsx:21]: In Popper 2 'Element' can't be passed?

File: uui-components/src/overlays/Tooltip.tsx [hunk 5]
```diff
@@ -129,6 +130,25 @@ export class Tooltip extends React.Component<TooltipProps, TooltipState> {
         </div>;
     }
 
+    private renderChildren = ({ ref, placement, style, arrowProps, isReferenceHidden }: PopperChildrenProps) => {
```
Comment (AlekseyManetov) [inline uui-components/src/overlays/Tooltip.tsx:133]: Maybe gives more specific name, like 'renderTooltipBody'?

File: uui-components/src/overlays/Tooltip.tsx [hunk 6]
```diff
@@ -129,6 +130,25 @@ export class Tooltip extends React.Component<TooltipProps, TooltipState> {
...
+        if (isReferenceHidden && this.state.isOpen) {
```
Comment (AlekseyManetov) [inline uui-components/src/overlays/Tooltip.tsx:134]: Please add comment : 
// Yes, we know that it's hack and we can perform setState in render, but we don't have other way to do it in this case

**Review by AlekseyManetov — CHANGES_REQUESTED** [review]

File: uui-components/src/overlays/Dropdown.tsx (hunk 1 above)
Comment (dsv-sergey) [reply uui-components/src/overlays/Dropdown.tsx:43]: image
Boundary is typed as HTMLElement, which inherits from Element

File: uui-components/src/overlays/Tooltip.tsx (hunk 4 above)
Comment (dsv-sergey) [reply uui-components/src/overlays/Tooltip.tsx:21]: Boundary is typed as HTMLElement, which extends from Element

**Review by AlekseyManetov — APPROVED** [review]

**Pull request merged**